lib/pymedphys/_experimental/streamlit/utilities/iview/ui.py

```diff
@@ -0,0 +1,341 @@
+# Copyright (C) 2021 Cancer Care Associates
+# Copyright (C) 2020 Cancer Care Associates and Simon Biggs
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import List, Union, cast
+
+from typing_extensions import Literal
+
+Number = Union[float, int]
+import datetime
+
+from pymedphys._imports import altair as alt
+from pymedphys._imports import numpy as np
+from pymedphys._imports import pandas as pd
+from pymedphys._imports import scipy
+from pymedphys._imports import streamlit as st
+
+from pymedphys._experimental.streamlit.utilities import icom as _icom
+from pymedphys._experimental.streamlit.utilities import iteration as _iteration
+
+from . import _angles, _filtering, _frames, _sync, _utilities
+
+
+def iview_and_icom_filter_and_align(
+    config, advanced_mode, filter_angles_by_default=False
+):
+    refresh_cache = st.button("Re-query databases")
+    (
+        database_directory,
+        icom_directory,
+        qa_directory,
+        database_table,
+        selected_date,
+        selected_machine_id,
+    ) = _utilities.get_directories_and_initial_database(config, refresh_cache)
+
+    icom_patients_directory = icom_directory.joinpath("patients")
+
+    database_table = _get_user_image_set_selection(database_table, advanced_mode)
+    database_table = _load_image_frame_database(
+        database_directory, database_table, refresh_cache, advanced_mode
+    )
+
+    filepaths_to_load, offset_to_apply = _sync.icom_iview_timestamp_alignment(
+        database_table,
+        icom_patients_directory,
+        selected_date,
+        selected_machine_id,
+        advanced_mode,
+    )
+
+    icom_datasets = []
+    for filepath in filepaths_to_load:
+        icom_dataframe = _icom.get_icom_dataset(filepath)
+        icom_datasets.append(icom_dataframe.copy())
+
+    icom_datasets = pd.concat(icom_datasets, axis=0, ignore_index=True)
+    icom_datasets = icom_datasets.sort_values(by="datetime", inplace=False)
+
+    icom_datasets["datetime"] = icom_datasets["datetime"] + datetime.timedelta(
+        seconds=offset_to_apply
+    )
+    icom_datasets["time"] = icom_datasets["datetime"].dt.round("ms").dt.time
+
+    try:
+        icom_datasets = _angles.make_icom_angles_continuous(icom_datasets)
+    finally:
+        if advanced_mode:
+            beam_on_mask = _utilities.expand_border_events(
+                np.diff(icom_datasets["meterset"]) > 0
+            )
+            beam_shade_min = -200 * beam_on_mask
```

Comment:
Magic number.
Create a constant with value 200 and use explanatory name

Reply:
Addressed in:

[`18dc8a1` (#1416)](https://github.com/pymedphys/pymedphys/pull/1416/commits/18dc8a13b3bfb0a56911dd79515717cd5d15fca1)